### ISSUE TITLE
Trend Forecasts

### ISSUE THREAD

**Issue by joaocunha1991** [facebook/prophet#560]
Hi, 

How is the trend extrapolated from the history to make a forecast? We have a user case where we have to run forecasts every month, and so, the model will be re-trained every month with a new extra month. 

We are tuning the prior.scales with grids and observing the performance on unseen data (in this case the last observed 4 Months). 

We noticed that the point forecasts from month to month vary a lot (in terms of trend) when re-training the model with the same prior.scales. The model seems to extrapolate the recent observed localised trend in a constant way. Is this the case?

If this is the case, how could we minimise this variation without significantly lose test set accuracy (MAPE). Is it a good idea to tune the prior.scales and impose a stricter cap on the changepoint.prior.scale parameter (to avoid overfitting the trend)?

Thanks !
Joao

**Comment by bletham** [comment]
This is correct, the trend that is projected forward will be that of the last segment. Frequent trend changes in the past will be represented by wide uncertainty intervals on that future trend.

By default Prophet will not insert changepoints in the last 20% of the time series, as a heuristic to avoid overfitting to small trend changes at the end, but if you're still seeing more variability there than you'd like, I would definitely start by substantially lowering the changepoint.prior.scale. You could use the cross_validation function to see what provides the right balance between being flexible enough to learn the seasonality, while not overfitting.
